src/udev/udev-builtin-net_id.c

```diff
@@ -345,6 +345,23 @@ static int dev_pci_slot(sd_device *dev, struct netnames *names) {
                 if (sd_device_get_sysname(hotplug_slot_dev, &sysname) < 0)
                         continue;
 
+                /* match slot address to function_id, if available */
+                if (naming_scheme_has(NAMING_SLOT_FUNCTION_ID) &&
+                    sd_device_get_sysattr_value(hotplug_slot_dev, "function_id", &attr) >= 0) {
+                        int function_id;
+                        char str[PATH_MAX];
```

Comment:
we try to avoid allocating overly large strings like this.

Please use asprintf() and allocate things on the heap

Reply:
Done.